### PR TITLE
More view tests and hoc.js bugs

### PR DESCRIPTION
**Pull request by yusufsafak**
@christianalfoni  When I was trying to improve test coverage of hoc.js, I found some bugs. I fixed small ones, but I don't have time to fix one of the bug related to computed.
Computeds are not automatically removed before a component is unmounted (componentWillUnmount). There is a problem at https://github.com/cerebral/cerebral/blob/master/packages/cerebral/src/viewFactories/Hoc.js#L70

### CODE REVIEW

File: packages/cerebral/src/viewFactories/react.test.js [hunk 1]
```diff
@@ -136,10 +137,182 @@ describe('React', () => {
...
+        return (
+          <div>{props.foo}</div>
+        )
+      })
+      const tree = TestUtils.renderIntoDocument((
+        <Container controller={controller}>
+          <TestComponent />
+        </Container>
+      ))
+      // how to get hasWarnedBigComponent or console.log??
```
Comment (yusufsafak) [inline packages/cerebral/src/viewFactories/react.test.js]: How can we test hasWarnedBigComponent? should we test console outputs?

**Comment by christianalfoni** [comment]
@yusufsafak Great stuff!

About warning, you can override `console.warn` temporarily for the test and check if it is called :)

About the computed bug, you have some more info on that? I do not see where there is an issue?

**Comment by Guria** [comment]
A bit of linting issues should be fixed

**Comment by yusufsafak** [comment]
Sorry guys for late reply. I was in a hurry to go holiday :) I pushed it so late and I couldn't explain it well. 
I am on mobile phone now and there is no good connection:( I just commented out two console tests to fix the lint errors. 
For the computed bug you can check this test https://github.com/cerebral/cerebral/blob/cb852e91d1b10dc098972530dec2a9a92c4d6d0c/packages/cerebral/src/viewFactories/react.test.js#L225 and coverage. Unmounting the component has never touch this line https://coveralls.io/builds/9429704/source?filename=packages%2Fcerebral%2Fsrc%2FviewFactories%2FHoc.js#L63 because there is no computed instance in depsmap. https://github.com/cerebral/cerebral/blob/master/packages/cerebral/src/viewFactories/Hoc.js#L70. It has only computed's depsmap.

**Comment by christianalfoni** [comment]
Looks great! Let me have a check on those commented out tests

**Comment by Guria** [comment]
@christianalfoni any news?

**Comment by Guria** [comment]
@christianalfoni revisit this please

**Comment by christianalfoni** [comment]
@Guria I looked into the branch and there has been so many changes to HOC that I think we need to throw this PR. A lot of the tests was related to previous signature, Computed etc. :(

Really sorry @yusufsafak!

**Comment by yusufsafak** [comment]
No problem!
Thanks for the new Computed and the other changes. They are really good :)

**Pull request closed without merging**